packages/ketcher-polymer-editor-react/src/components/shared/modal/Modal.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton
+} from '@mui/material'
+import React, { useMemo } from 'react'
+import { useTheme } from '@emotion/react'
+import styled from '@emotion/styled'
+import { Icon } from 'components/shared/icon'
+import { scrollbarThin } from 'styles/mixins'
+
+interface ModalProps {
+  children: JSX.Element | Array<JSX.Element>
+  title: string
+  isOpen: boolean
+  showCloseButton?: boolean
+  onClose?: () => void
+}
+
+const Header = styled(DialogTitle)(({ theme }) => ({
+  margin: '12px 11px 12px 16px',
+  padding: 0,
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  fontFamily: `${theme.font.family.inter}`,
+  fontSize: `${theme.font.size.medium}`,
+  fontWeight: `${theme.font.weight.regular}`,
+  textTransform: 'uppercase'
+}))
+
+const Title = styled.div({
+  marginRight: '10px'
+})
+
+const Content = styled(DialogContent)(({ theme }) => ({
+  margin: '0 16px 16px',
+  padding: 0,
+
+  ...scrollbarThin(theme)
+}))
+
+const Footer = styled(DialogActions)({
+  margin: '0 16px 16px',
+  padding: 0
+})
+
+type ModalSubcomponent = 'Content' | 'Footer'
+
+export const Modal = ({
+  children,
+  title,
+  isOpen,
+  showCloseButton = true,
+  onClose
+}: ModalProps) => {
+  const theme = useTheme()
+
+  const paperProps = useMemo(
```

Comment:
I don't think you need useMemo here

Reply:
I used useMemo to memoize this object creation, because I pass this object to the Dialog's props